plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/toolwindow/AmazonQToolWindowFactory.kt

```diff
@@ -22,20 +22,31 @@ import software.aws.toolkits.jetbrains.core.credentials.pinning.QConnection
 import software.aws.toolkits.jetbrains.core.credentials.sono.Q_SCOPES
 import software.aws.toolkits.jetbrains.core.credentials.sso.bearer.BearerTokenAuthState
 import software.aws.toolkits.jetbrains.core.credentials.sso.bearer.BearerTokenProviderListener
+import software.aws.toolkits.jetbrains.core.notifications.NotificationFollowupActions
+import software.aws.toolkits.jetbrains.core.notifications.ShowCriticalNotificationBannerListener
+import software.aws.toolkits.jetbrains.core.notifications.getBannerActionList
 import software.aws.toolkits.jetbrains.core.webview.BrowserState
 import software.aws.toolkits.jetbrains.services.amazonq.QWebviewPanel
 import software.aws.toolkits.jetbrains.services.amazonq.RefreshQChatPanelButtonPressedListener
 import software.aws.toolkits.jetbrains.services.amazonq.gettingstarted.openMeetQPage
 import software.aws.toolkits.jetbrains.utils.isQConnected
 import software.aws.toolkits.jetbrains.utils.isQExpired
 import software.aws.toolkits.jetbrains.utils.isQWebviewsAvailable
+import software.aws.toolkits.jetbrains.utils.notifyInfo
 import software.aws.toolkits.resources.message
 import software.aws.toolkits.telemetry.FeatureId
 import java.awt.event.ComponentAdapter
 import java.awt.event.ComponentEvent
+import javax.swing.JComponent
 
 class AmazonQToolWindowFactory : ToolWindowFactory, DumbAware {
+    private val notificationPanel = NotificationPanel()
+    private val qPanel = QPanel()
```

Comment:
toolwindowfactory is application-global so these cannot be initialized here

Reply:
also you don't want to be potentially creating panels on background anyways

Reply:
If we dont make these application wide, they are not initialized when the toolwindow is closed

Reply:
Discussed offline, maintaining state of notifications to show